mautic/mautic
# Capture SMS errors and add to the timeline

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

From the code, it looked like we were sending the sms twice. Also, if the SMS provider rejected, it was unknown why since the exception message was not captured. This PR fixes these two issues. The PR is dependent on https://github.com/mautic/mautic/pull/3219 which happens to include the needed changes to display failed entries in the timeline. 

#### Steps to test this PR:
1. Configure Twilio and send a SMS through a campaign. Purposively use a bad number. View the contact's timeline. The error message should display in the timeline entry.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1.  Repeat the above but this time nothing is recorded so it's unknown what happened.
